lib/Quantization/Base/Base.cpp

```diff
@@ -201,6 +201,27 @@ QuantizationTransform32To8 quantizeScaleOffset32To8(float scale,
   int preShift = 0;
   int postShift = 0;
 
+  // We treat first the particular case when scale is a power of 2 (2 ^ exp,
+  // where exp is a signed integer exponent). The operation is specialized as:
+  // - for positive 2's exponent:
+  //     x * scale + offset (pre = 0, post = 0, scale = (int)scale).
+  // - for negative 2's exponent:
+  //     x >> post + offset (pre = 0, post = -exp, scale = 1).
+  if (isFloatPowerOf2(scale)) {
+    int exp = getFloat2Exp(scale);
```

Comment:
Just want to double check, if "SymmetricWithPower2Scale" is not selected, should this flow be executed? Thanks!

Reply:
Yes.
- When the scale is a power of 2 (whether it was forced using the "SymmetricWithPower2Scale" schema or obtained by coincidence for other schema) it always makes sense to approximate the multiplication using only one operation (up/down scale).
- The previous logic, for example, for a scale = (exactly) 0.125, it yielded the following expression: x * 0.125 -> (x * 256) >> 11 (integer_scale = 256 and postShift=11), whereas it makes more sense to do just x >> 3 (integer_scale = 1 and postShift = 3).

As a side note, I don`t think the logic of this function (quantizeScaleOffset32To8) is rigorous enough (mathematically). Choosing the pre/post/scale should also take into account: data dynamic range in the quantized domain, the bit-width of the intermediate computation (accumulator), etc. But this will be the subject of a future PR.

Thanks!